common/spec/dependabot/pull_request_creator/github_spec.rb

```diff
@@ -612,6 +612,41 @@
       end
     end
 
+    context "when the branch already exists" do
+      before do
+        service_pack_response.gsub!("heads/rubocop", "heads/#{branch_name}")
+        Dependabot::Experiments.register(:dedup_branch_names, true)
+      end
+
+      after do
+        Dependabot::Experiments.register(:dedup_branch_names, false)
+      end
+
+      context "when a PR to this branch doesn't exist" do
+        before do
+          url = "#{repo_api_url}/pulls?head=gocardless:#{branch_name}" \
+                "&state=all"
+          stub_request(:get, url)
+            .to_return(status: 200, body: "[]", headers: json_header)
+          stub_request(
+            :patch,
+            "#{repo_api_url}/git/refs/heads/#{branch_name}"
+          ).to_return(
+            status: 200,
+            body: fixture("github", "update_ref.json"),
+            headers: json_header
+          )
+        end
+
+        it "returns a suitable exception" do
+          expect { creator.create }
+            .to raise_error(Dependabot::PullRequestCreator::DuplicateBranchExists,
```

Comment:
I think it makes more sense if we call this error `BranchAlreadyExists` since the branch we are trying to make is the duplicate.